### PR TITLE
feat(logout): handle session expiration (AR-1556)

### PR DESCRIPTION
**Pull request by mchenani**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1556" title="AR-1556" target="_blank">StoryAR-1556</a>  Admin deletes a group conversation for everyone in the group
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
This PR cover 3 topics around user session:
- Handle remove client event: when a user delete one of their clients on other client, all other clients will receive client_removed event with the clientId.
- Handle user account deletion: when a user delete its account from other clients, all other clients will receive user_delete event.
In both above events + the selfLogout we need to send the reason of the logout to the AR app! So I had to refactor the AuthSession to support that idea.

### Issues
- Handle user account deletion
- Handle client remove
- Send the logout/session expiration reason to the AR

### Causes (Optional)
The events weren't handled before.

### Solutions

Handle the events in the userEventsReceiver and refactor the AuthSession

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:
Needs to release Kalium related PR as well.
https://github.com/wireapp/kalium/pull/661
- [x] GitHub link to other pull request

#### Test Coverage (Optional)
WIP
- [ ] I have added automated test to this contribution

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/12/) **failed**.

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/13/) **failed**.

**Comment by AndroidBob** [comment]
Build [14](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/14/) **failed**.

File: app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/SummaryNotificationDismissReceiver.kt [hunk 1]
```diff
@@ -39,7 +38,7 @@ class SummaryNotificationDismissReceiver : BroadcastReceiver() {
                 } else {
                     val currentSession = coreLogic.globalScope { session.currentSession() }
                     if (currentSession is CurrentSessionResult.Success) {
-                        coreLogic.getSessionScope(currentSession.authSession.tokens.userId)
+                        coreLogic.getSessionScope(currentSession.authSession.session.userId)
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/SummaryNotificationDismissReceiver.kt:41]: I'm just being nitpicky, but I feel like the name 'authSession.session' is wrong 😅
maybe we can revert it back to tokens

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
🍇 work 🚀, only small changes needed to remove unnecessary assisted injection

**Comment by AndroidBob** [comment]
Build [16](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/16/) **failed**.

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [17](https://10.10.124.134/job/wire-android-reloaded/job/PR-777/17/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4038810)(4038810)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4038810)(4038810)-dev-release.apk)

**Pull request merged**